src/kibana/components/vislib/lib/handler/types/pie.js

```diff
@@ -10,10 +10,8 @@ define(function (require) {
      */
 
     return function (vis) {
-      var data = new Data(vis.data, vis._attr);
-
       return new Handler(vis, {
-        legend: new Legend(vis, vis.el, data.pieNames(), data.getPieColorFunc(), vis._attr),
+        legend: new Legend(vis, vis.data),
```

Comment:
Any reason why the Legend can't just pull data from the vis? Why pass it in separately?